lib/lambda_ethereum_consensus/validator/proposer.ex

```diff
@@ -2,17 +2,42 @@ defmodule LambdaEthereumConsensus.Validator.Proposer do
   @moduledoc """
   Validator proposer duties.
   """
+
+  alias LambdaEthereumConsensus.Execution.ExecutionChain
+  alias LambdaEthereumConsensus.Execution.ExecutionClient
   alias LambdaEthereumConsensus.P2P.Gossip.OperationsCollector
   alias LambdaEthereumConsensus.StateTransition
   alias LambdaEthereumConsensus.StateTransition.Accessors
   alias LambdaEthereumConsensus.StateTransition.Misc
+  alias LambdaEthereumConsensus.Store.Blocks
+  alias LambdaEthereumConsensus.Store.BlockStates
   alias LambdaEthereumConsensus.Utils.BitVector
+  alias LambdaEthereumConsensus.Utils.Randao
   alias LambdaEthereumConsensus.Validator.BlockRequest
 
   alias Types.BeaconBlock
   alias Types.BeaconState
   alias Types.SignedBeaconBlock
 
+  def build_block(%BlockRequest{} = request) do
+    parent_root = request.parent_root
+    proposed_slot = request.slot
+    pre_state = BlockStates.get_state!(parent_root)
+
+    with {:ok, execution_payload} <-
+           pre_state
+           |> StateTransition.process_slots(proposed_slot)
+           |> build_execution_block(parent_root) do
+      block_request =
+        request
+        |> Map.merge(fetch_operations_for_block())
+        |> Map.put(:eth1_data, fetch_eth1_data(proposed_slot, pre_state))
+        |> Map.put(:execution_payload, execution_payload)
+
+      construct_block(pre_state, block_request, request.privkey)
```

Comment:
Since we already have the `privkey` in the `block_request`, we should drop it from the args.